flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/deployment/ApplicationReconciler.java

```diff
@@ -195,6 +201,34 @@ protected void deploy(
                 relatedResource.getMetadata(), spec, deployConfig, kubernetesClient);
     }
 
+    private void setJobIdIfNecessary(
+            FlinkDeploymentSpec spec, FlinkDeploymentStatus status, Configuration deployConfig) {
+        // https://issues.apache.org/jira/browse/FLINK-19358
+        // https://issues.apache.org/jira/browse/FLINK-29109
+        if (spec.getFlinkVersion().isNewerVersionThan(FlinkVersion.v1_15)) {
+            return;
+        }
+
+        final JobID jobId;
```

Comment:
We could use the String representation and set this directly from `status.getJobStatus().getJobId()` to get rid of the `else` branch later.